library/core/src/fmt/num.rs

```diff
@@ -236,10 +216,24 @@ macro_rules! impl_Display {
             }
         }
 
+        #[stable(feature = "rust1", since = "1.0.0")]
+        impl fmt::Display for $signed {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                #[cfg(not(feature = "optimize_for_size"))]
+                {
+                    return self.unsigned_abs()._fmt(*self >= 0, f);
+                }
+                #[cfg(feature = "optimize_for_size")]
+                {
+                    return $gen_name((!self.unsigned_abs().$conv_fn()), *self >= 0, f);
```

Comment:
Wait, I think the `!` shouldn't be there.

Reply:
whoops.